### PR TITLE
Create codescanning workflow

### PR DESCRIPTION
**Pull request by srt32**
This PR adds a code scanning workflow for javascript (ruby is not yet supported). Shall we experiment with running this workflow for a bit?

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'ruby', 'javascript' ]
```
Comment (srt32) [inline .github/workflows/codeql-analysis.yml:20]:
```suggestion
        language: [ 'javascript' ]
```

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
```
Comment (manuelpuyol) [inline .github/workflows/codeql-analysis.yml:5]: is this the correct syntax?

Comment (srt32) [reply .github/workflows/codeql-analysis.yml:5]: I think I pulled this from the sample config they gave me so I think so?!

**Review by manuelpuyol — APPROVED** [review]
not an expert on this, but looks good

**Pull request merged**